orchagent/routeorch.cpp

```diff
@@ -649,6 +659,16 @@ bool RouteOrch::addRoute(IpPrefix ipPrefix, IpAddresses nextHops)
 
     m_syncdRoutes[ipPrefix] = nextHops;
 
+    /**
+     * If a route is added to m_syncdRoutes and was added to m_syncdTempRoutes,
+     * remove this entry from the m_syncdTempRoutes.
+     */
+    if (m_syncdTempRoutes.find(ipPrefix) != m_syncdTempRoutes.end() &&
```

Comment:
first part is not enough. it is to check if the value is the same and then remove the entry. because no matter temp routes or final routes, they all go to here.

---
In reply to: [119508324](https://github.com/Azure/sonic-swss/pull/230#discussion_r119508324) [](ancestors = 119508324)